01_fib/nicholsk/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+  fib(7)
+}
+
+func fib(n int) {
+
+	if(n < 0){
+		fmt.Fprintln(out, "fib(n) doesn't accept negative integers")
+		return
+	}
+
+	if(n == 0){
+		fmt.Fprintln(out, n)
+		return
+	}
+
+	current, previous := 1,0
+
+	// print first value which is always 1
+	fmt.Fprintln(out, current)
+
+	for i := 1; i < n; i++ {
+
+		// calculate next number in sequence and print out
+		sum := previous + current
+		fmt.Fprintln(out, sum)
+
+		// update variables for next iteration
+		previous = current
```

Comment:
You can use multiple assignment here without worrying about overwriting
```
previous, current = current, sum
```

You can actually use this before the print to remove the sum variable.

Reply:
Neat! Thanks Dan 👍